File: java/exercises/01-beginner/ex032_inheritance_override/InheritanceOverride.java
package fewolearning.exercises.beginner.ex032_inheritance_override;

/*
Exercise 032 - Inheritance override (beginner).

Goal:   Override a base method and extend its behavior using super.
Drills: inheritance, overriding, super.
*/
public final class InheritanceOverride {
    private InheritanceOverride() {
    }

    public static class Animal {
        public String describe() {
            return "an animal";
        }
    }

    public static class Dog extends Animal {
        @Override
        public String describe() {
            throw new UnsupportedOperationException("TODO");
        }
    }
}

File: kotlin/exercises/01-beginner/ex024_inheritance_override/InheritanceOverride.kt
package fewolearning.exercises.beginner.ex024_inheritance_override

/*
Exercise 024 - Inheritance override (beginner).

Goal:   Override a base method and extend its behavior using super.
Drills: open classes, overriding, super.
*/
open class Animal {
    open fun describe(): String = "an animal"
}

class Dog : Animal() {
    override fun describe(): String {
        TODO()
    }
}
